File: app/src/main/java/io/euphoria/xkcd/app/data/Message.java
package io.euphoria.xkcd.app.data;

/* Created by Xyzzy on 2017-02-24. */

/* A single Heim message */
public interface Message {

    /* The ID of this message as a string */
    String getID();

    /* The parent of this message (null for none) */
    String getParent();

    /* The timestamp of this message, as milliseconds since Epoch */
    long getTimestamp();

    /* The author of the message */
    SessionView getSender();

    /* The content of the message */
    String getContent();

    /* Whether the message was truncated to save bandwidth */
    boolean isTruncated();

}

File: app/src/main/java/io/euphoria/xkcd/app/data/Message.kt
package io.euphoria.xkcd.app.data

/** Created by Xyzzy on 2017-02-24.  */ /* A single Heim message */
interface Message {
    /* The ID of this message as a string */
    val id: String

    /* The parent of this message (null for none) */
    val parent: String?

    /* The timestamp of this message, as milliseconds since Epoch */
    val timestamp: Long

    /* The author of the message */
    val sender: SessionView

    /* The content of the message */
    val content: String

    /* Whether the message was truncated to save bandwidth */
    val isTruncated: Boolean
}
